src/directives/column-reorder.js

```diff
@@ -0,0 +1,223 @@
+                        var rowTemplate = templateMapper[rowId];
+                        rowTemplate = angular.element('<tr>' + rowTemplate + '<tr>');
+                        rowTemplate = angular.element(rowTemplate[0].children);
+
+                        var columnCount = rowTemplate.length;
+                        if (oldColIndex >= columnCount || newColIndex >= columnCount) {
+                            return;
+                        }
+
+                        if (offset < 0) {
+                            // Insert Before
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                        } else if (offset > 0) {
+                            // Insert After
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex + 1, 0, oldColumn);
+                        } else {
+                            // Swap Column Places
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            var newColumn = rowTemplate.splice(newColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                            rowTemplate.splice(oldColIndex, 0, newColumn);
+                        }
+
+                        var newRowTemplate = '';
+                        for(var i = 0; i < rowTemplate.length; i++) {
+                            newRowTemplate += rowTemplate[i].outerHTML;
+                        }
+
+                        templateMapper[rowId] = newRowTemplate;
+                    });
+
+                    $scope.$broadcast('drawTableRows');
+                }
+
+            },
+            compile: function($element, $attrs) {
+                var headerRow = $element[0].querySelector('thead tr');
+                headerRow.setAttribute('reorderable-column-template-mapper', getId());
+
+                var headers = headerRow.querySelectorAll('th');
+                var headerCount = headers.length;
+                while (headerCount--) {
+                    headers[headerCount].setAttribute('reorderable-header-drag-drop', '');
+                }
+
+                var bodyRows = $element[0].querySelectorAll('tbody tr');
+                var bodyRowCount = bodyRows.length;
+                while (bodyRowCount--) {
+                    bodyRows[bodyRowCount].setAttribute('reorderable-column-template-mapper', getId());
+                }
+
+            }
+        }
+    });
+
+    module.directive('reorderableColumnTemplateMapper', function($compile) {
+        return {
+            require: '^reorderableColumns',
+            compile: function($element, $attrs) {
+                var directiveName = this.name;
+                var rowId = $attrs[directiveName];
+
+                var uncompiledHtml = $element[0].innerHTML;
```

Comment:
Right, the hidden column is being set by a CSS class, if is exists before in the uncompiled template then it will always be there, or the opposite where the a hidden column will become unhidden.
It seems like the hiding of the columns should be passed on a model value of the row or something that the column hiding function can manage and reapply after the rows change.